### ISSUE TITLE
agent: promtail causing very high cpu load when running and stops

### ISSUE THREAD

**Issue by chenjpu** [grafana/agent#1820]
The high CPU load starts suddenly and disappears after restarting agent.

![image](https://user-images.githubusercontent.com/1785778/175220455-c3e833d0-816d-4288-a4c5-de9e3c7f3781.png)

![1655961524997](https://user-images.githubusercontent.com/1785778/175220893-60d75646-6dfd-4fff-8c58-c0c6d7a4c81e.png)

[from promtail ](https://github.com/grafana/loki/issues/5350)

**Comment by tpaschalis** [comment]
Hey there 👋  Thanks for the report!

I read through the linked Loki issue, are you encountering the same problems? That is, high CPU usage when a workflow stops, or does it happen at a random moment?

We currently haven't ported the commit that solved the linked issue, but I think it would be easy to do so and see if it makes any difference.

Also, since it seems you're comfortable with pprof, would you mind generating a 29-second CPU profile when the issue appears and uploading the `.gz` files so we could take a look ourselves as well?

**Comment by chenjpu** [comment]
The phenomenon found at present is that when the service starts and stops, but also the occasional phenomenon.

[profile.gz](https://github.com/grafana/agent/files/8983547/profile.gz)

NOTE:
During the troubleshooting process, the possible reason is that two different services listen to the same file path due to incorrect configuration

**Comment by chenjpu** [comment]
This problem is reproduced locally
[profile.gz](https://github.com/grafana/agent/files/8987862/profile.gz)

ENV：nomad + consul

config:

```
     - job_name: consul
       consulagent_sd_configs:
       - server: '127.0.0.1:8500'
       relabel_configs:
         - source_labels: [__meta_consulagent_tags]
           regex: .*,(loki|dapr),.*
           action: keep
         #1- source_labels: [__meta_consulagent_service_metadata_ALLOC_ID]
         #2  regex:  \w{8}-\w{4}-\w{4}-\w{4}-\w{12}
         #3  action: keep
         - source_labels: [__meta_consulagent_service]
           target_label: service
         - source_labels: [__meta_consulagent_node]
           target_label: node     
         - source_labels: [__meta_consulagent_service_metadata_env]
           target_label:  env
         - source_labels: [__meta_consulagent_service_metadata_app_code]
           target_label:  app_code
         - source_labels: [__meta_consulagent_service_metadata_ALLOC_ID]
           target_label:  __path__
           replacement:   '/opt/nomad/data/alloc/${1}/alloc/logs/[a-zA-Z]*'
       pipeline_stages:
          - labeldrop:
            - filename
```
setp
1.Turn off the 1,2,3 configuration
2.Scroll to upgrade a service with a missing alloc_id meta

**Comment by rfratto** [comment]
For maintainers: this will be fixed when #1930 is addressed, since the latest release of Loki contains the fix for this issue.
